### ISSUE TITLE
[Issue]: The app is not working

### ISSUE THREAD

**Issue by Meshary707** [SamSprung/SamSprung-TooUI#218]
### Issue Description

The app hasn't been working for a while. 

### Logcat Output

```shell
SamSprung (github) #07b259a
Android S_V2 (12)

java.lang.RuntimeException: Unable to start activity ComponentInfo{com.eightbit.samsprung/com.eightbit.samsprung.settings.CoverPreferences}: java.lang.StringIndexOutOfBoundsException: length=0; index=0
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4237)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4403)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:104)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2498)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8855)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
Caused by: java.lang.StringIndexOutOfBoundsException: length=0; index=0
	at java.lang.String.charAt(Native Method)
	at com.eightbit.samsprung.settings.FilteredAppsAdapter.getSections(FilteredAppsAdapter.kt:66)
	at myinnos.indexfastscrollrecycler.IndexFastScrollRecyclerView.setAdapter(IndexFastScrollRecyclerView.java:21)
	at com.eightbit.samsprung.settings.CoverPreferences.onCreate(CoverPreferences.kt:2315)
	at android.app.Activity.performCreate(Activity.java:8326)
	at android.app.Activity.performCreate(Activity.java:8306)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1347)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4212)
	... 12 more
```

### Guide Confirmation

- [X] I have reviewed the guide and included valid logcat content

**Comment by AbandonedCart** [comment]
Unfortunately, there is not much that can be done about how long the app wasn't working before it was reported. We don't usually know it isn't working for someone until they tell us. 

It appears the issue is an invalid app name causing the sorting to crash. A check for such things will be added.

**Comment by AbandonedCart** [comment]
Fixed by a5a8e977a962ab4ef9d04abf50da85b4128f6b26
